src/pke/lib/scheme/ckksrns/ckksrns-cryptoparameters.cpp

```diff
@@ -154,4 +159,27 @@ uint64_t CryptoParametersCKKSRNS::FindAuxPrimeStep() const {
     return static_cast<uint64_t>(2 * n);
 }
 
+void CryptoParametersCKKSRNS::ConfigureCompositeDegree(uint32_t scalingModSize) {
+    // Add logic to determine whether composite scaling is feasible or not
+    if (GetScalingTechnique() == COMPOSITESCALINGAUTO) {
+        uint32_t registerWordSize = GetRegisterWordSize();
+        if (registerWordSize <= 64) {
+            if (registerWordSize < scalingModSize) {
+                uint32_t compositeDegree =
+                    static_cast<uint32_t>(std::ceil(static_cast<float>(scalingModSize) / registerWordSize));
+                // Assert minimum allowed moduli size on composite scaling mode
+                // @fdiasmor TODO: make it more robust for a range of multiplicative depth
```

Comment:
is 22 going to stay as a hardcoded value or it can relate to modsize?

Reply:
@dsuponitskiy At this stage, it will stay hardcoded because 22 was found experimentally/heuristically. My plan is to try to create a lookup table for this value based on the multiplicative depth; however, for now this value works well for most practical cases. Otherwise, one can always resort to COMPOSITESCALINGMANUAL scaling technique.

Reply:
This is reasonable. It depends on the ring dimension and does not depend on NATIVE_SIZE.

Reply:
ok